include/h2o/hostinfo.h

```diff
@@ -65,7 +65,7 @@ int h2o_hostinfo_aton(h2o_iovec_t host, struct in_addr *addr);
 
 /* inline defs */
 
-inline struct addrinfo *h2o_hostinfo_select_one(struct addrinfo *res)
+inline struct addrinfo *h2o_hostinfo_select_one(struct addrinfo *res, uint32_t addrinfo_sel)
```

Comment:
How about using `size_t *` for the selector? Reasons:
* we can use `NULL` to specify the random behavior without having an off-by-one adjustment
* `size_t` is sensible than `uint32_t`, because we'd have the guarantee that the maximum number of objects returned by a linked-list of pointers would be smaller than that (assuming a flat memory model)

Reply:
Fixed